test/integration/driver.go

```diff
@@ -86,58 +74,94 @@ func main() {
 	log.Printf("hub %v, tag %v", hub, tag)
 
 	// connect to k8s
-	client = connect()
+	var err error
+	client, err = connect()
+	if err != nil {
+		log.Fatal(err)
+	}
 	if namespace == "" {
-		namespace = generateNamespace(client)
+		namespace, err = generateNamespace(client)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer func() {
+			deleteNamespace(client, namespace)
+		}()
 	}
 
 	// write template
 	f, err := os.Create(yaml)
-	check(err)
+	if err != nil {
+		log.Fatal(err)
+	}
 	w := bufio.NewWriter(f)
 
-	check(write("test/integration/manager.yaml.tmpl", map[string]string{
+	if err := write("test/integration/manager.yaml.tmpl", map[string]string{
 		"hub": hub,
 		"tag": tag,
-	}, w))
+	}, w); err != nil {
+		log.Fatal(err)
+	}
 
-	check(write("test/integration/http-service.yaml.tmpl", map[string]string{
+	if err := write("test/integration/http-service.yaml.tmpl", map[string]string{
 		"hub":   hub,
 		"tag":   tag,
 		"name":  "a",
 		"port1": "8080",
 		"port2": "80",
-	}, w))
+	}, w); err != nil {
+		log.Fatal(err)
+	}
 
-	check(write("test/integration/http-service.yaml.tmpl", map[string]string{
+	if err := write("test/integration/http-service.yaml.tmpl", map[string]string{
 		"hub":   hub,
 		"tag":   tag,
 		"name":  "b",
 		"port1": "80",
 		"port2": "8000",
-	}, w))
+	}, w); err != nil {
+		log.Fatal(err)
+	}
 
-	check(write("test/integration/external-services.yaml.tmpl", map[string]string{
+	if err := write("test/integration/external-services.yaml.tmpl", map[string]string{
 		"hub":       hub,
 		"tag":       tag,
 		"namespace": namespace,
-	}, w))
+	}, w); err != nil {
+		log.Fatal(err)
+	}
+
+	if err := w.Flush(); err != nil {
+		log.Fatal(err)
+	}
+	if err := f.Close(); err != nil {
+		log.Fatal(err)
+	}
 
-	check(w.Flush())
-	check(f.Close())
+	if err := run("kubectl apply -f " + yaml + " -n " + namespace); err != nil {
+		log.Fatal(err)
```

Comment:
Don't you have to create different echo.yaml s one per namespace? Atleast, that would be useful from a debugging perspective, when I ctrl+c during the test, to see whats up with the failures, etc.